apple/swift-docc
# Add page image directive

Bug/issue #, if applicable: rdar://97739580

## Summary

Add support for an `@PageImage` directive that allows for customizing a given page's card or icon image.

- https://forums.swift.org/t/supporting-more-types-of-documentation-with-swift-docc/59725#pageimage-22

## Dependencies

- https://github.com/apple/swift-docc-render/pull/417

## Testing

1. Build a copy of Swift-DocC-Render from https://github.com/apple/swift-docc-render/pull/417
2. Add an SVG with an ID to a DocC catalog: [plus.svg](https://user-images.githubusercontent.com/6750147/190448278-44f576f4-7a7d-4d04-995c-7481a129f47d.svg)
3. Add a page image metadata directive to a documentation extension file or article:

  ```md
  @Metadata {
    @PageImage(source: "plus", purpose: icon)
  }
  ```
4. Confirm that the icon appears in the navigator and hero. Confirm that there are references to the icon on any page that references the customized page, the page itself, and the index.json.

<img width="1695" alt="Screenshot 2022-09-15 at 4 50 06 PM" src="https://user-images.githubusercontent.com/6750147/190449714-bdbf25f1-e97f-43c7-9c43-b317416d1677.png">


## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
